mmpose/apis/inference.py

```diff
@@ -333,38 +345,42 @@ def inference_top_down_pose_model(model,
     pose_results = []
     returned_outputs = []
 
-    with OutputHook(model, outputs=outputs, as_tensor=False) as h:
-        for person_result in person_results:
-            if format == 'xyxy':
-                bbox_xyxy = np.expand_dims(np.array(person_result['bbox']), 0)
-                bbox_xywh = _xyxy2xywh(bbox_xyxy)
-            else:
-                bbox_xywh = np.expand_dims(np.array(person_result['bbox']), 0)
-                bbox_xyxy = _xywh2xyxy(bbox_xywh)
-
-            if bbox_thr is not None:
-                assert bbox_xywh.shape[1] == 5
-                if bbox_xywh[0, 4] < bbox_thr:
-                    continue
-
-            pose, heatmap = _inference_single_pose_model(
-                model,
-                img_or_path,
-                bbox_xywh[0],
-                dataset,
-                return_heatmap=return_heatmap)
+    # Change for-loop preprocess each bbox to preprocess all bboxes at once.
+    bboxes = np.array([box['bbox'] for box in person_results])
+
+    # Select bboxes by score threshold
+    if bbox_thr is not None:
+        assert bboxes.shape[1] == 5
```

Comment:
```
python demo/top_down_pose_tracking_demo_with_mmdet.py \
    demo/mmdetection_cfg/faster_rcnn_r50_fpn_coco.py \
http://download.openmmlab.com/mmdetection/v2.0/faster_rcnn/faster_rcnn_r50_fpn_1x_coco/faster_rcnn_r50_fpn_1x_coco_20200130-047c8118.pth \
    configs/top_down/resnet/coco/res50_coco_256x192.py \
    https://download.openmmlab.com/mmpose/top_down/resnet/res50_coco_256x192-ec54d7f3_20200709.pth \
    --video-path /dev/video0 \
    --show
```
**Exception handling is not exist when the human bounding boxes are not detected**
```
Use load_from_http loader
Use load_from_http loader
Traceback (most recent call last):
  File "demo/top_down_pose_tracking_demo_with_mmdet.py", line 177, in <module>
    main()
  File "demo/top_down_pose_tracking_demo_with_mmdet.py", line 132, in main
    pose_results, returned_outputs = inference_top_down_pose_model(
  File "/mmpose/mmpose/apis/inference.py", line 370, in inference_top_down_pose_model
    assert bboxes.shape[1] == 5
IndexError: tuple index out of range
```

Reply:
Thanks! Please open an issue so that it is easier to track progress